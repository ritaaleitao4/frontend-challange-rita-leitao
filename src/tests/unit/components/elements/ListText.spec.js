import Vue from 'vue'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import EditText from '@/components/elements/EditText.vue'

const localVue = createLocalVue()
let wrapper

describe('EditText Component Tests', () => {
  it('Snapshot test', () => {
    const Constructor = Vue.extend(EditText)
    const vm = new Constructor().$mount()
    expect(vm.$el).toMatchSnapshot()
  })

  it('should have values if title editable', () => {
    const value = 'Value of title'
    const type = ''

    wrapper = shallowMount(EditText, {
      localVue,
      propsData: { value, type },
    })

    expect(wrapper.find('span').text()).toBe(value)
    expect(wrapper.contains('input')).toBe(true)
    expect(wrapper.contains('textarea')).toBe(false)
  })

  it('should have values if textarea editable', () => {
    const value = 'Value of text'
    const type = 'textArea'

    wrapper = shallowMount(EditText, {
      localVue,
      propsData: { value, type },
    })

    expect(wrapper.find('span').text()).toBe(value)
    expect(wrapper.contains('input')).toBe(false)
    expect(wrapper.contains('textarea')).toBe(true)
  })

  it('should emit event on editing', () => {
    const value = 'Value'
    wrapper = shallowMount(EditText, {
      localVue,
      propsData: { value },
    })

    wrapper.find('input').trigger('input')
    expect(wrapper.emitted('input')).toBeTruthy()
  })

  it('should toogle active input', () => {
    const value = ''
    wrapper = shallowMount(EditText, {
      localVue,
      propsData: { value },
    })

    wrapper.find('span').trigger('click')
    expect(wrapper.contains('.hide')).toBe(false)

    wrapper.find('input').trigger('blur')
    expect(wrapper.contains('.hide')).toBe(false)
  })

  it('should toogle active input (textarea)', () => {
    const type = 'textArea'
    const value = 'Value of text'

    wrapper = shallowMount(EditText, {
      localVue,
      propsData: { type, value },
    })

    wrapper.find('span').trigger('click')
    expect(wrapper.contains('.hide')).toBe(false)

    wrapper.find('textarea').trigger('blur')
    expect(wrapper.contains('.hide')).toBe(false)
  })
})
