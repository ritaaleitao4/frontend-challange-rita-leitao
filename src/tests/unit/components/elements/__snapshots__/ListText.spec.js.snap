// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListText Component Tests Snapshot test 1`] = `
<div>
  <span
    class="empty"
  >
    
    Click to add voice
  
  </span>
   
  <input
    type="text"
  />
   
  <!---->
</div>
`;
